src/tooltip/tooltip.js

```diff
@@ -422,13 +423,67 @@ angular.module('ui.bootstrap.tooltip', ['ui.bootstrap.position', 'ui.bootstrap.s
               }
             }
 
+            // hide tooltips/popovers for clickaway trigger
+            function bodyHideTooltipBind(e) {
+              if (!ttScope.isOpen) {
+                return;
+              }
+              // get element that was clicked
+              var target_elem = angular.element(e.target);
+
+              // make sure the tooltip/popover link was not the element clicked (keep in mind, the click event on the link toggles show/hide already)
+              if (e.target != element[0]) {
+                // hide by default
+                var shouldHide = true;
+
+                // loop through all ttScopes in the stackedMap (openedTooltips)
+                var ottScopes = openedTooltips.keys();
+                for (var index in ottScopes) {
```

Comment:
Should consider using angular.forEach as that implements the hasOwnProperty check, [link](https://docs.angularjs.org/api/ng/function/angular.forEach#!)